test/gtest/ucp/test_ucp_ep_reconfig.cc

```diff
@@ -308,9 +369,44 @@ void test_ucp_ep_reconfig::skip_non_p2p()
     }
 }
 
+bool test_ucp_ep_reconfig::has_bond_iface()
+{
+    auto context = sender().ucph();
+    const ucp_tl_resource_desc_t *rsc;
+
+    ucs_carray_for_each(rsc, context->tl_rscs, context->num_tls) {
+        std::string tl_name(rsc->tl_rsc.dev_name);
+
+        if (tl_name.find("bond") != std::string::npos) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+void test_ucp_ep_reconfig::ensure_reused_lanes_reconfigurable()
+{
+    if (!reuse_lanes()) {
+        return;
+    }
+
+    if (has_transport("ud_v") || has_transport("ud_x")) {
+        UCS_TEST_SKIP_R("the test requires at least 2 lanes, while UD has only "
+                        "1");
+    }
+
+    if (has_transport("tcp") || has_transport("dc_x") || has_transport("shm")) {
+        UCS_TEST_SKIP_R("non wired-up lanes are not supported yet");
+    }
+
+    if (has_bond_iface()) {
+        modify_config("IB_NUM_PATHS", "1", SETENV_IF_NOT_EXIST);

```

Comment:
in some places in the code there is check for num_paths, why not set "IB_NUM_PATHS" to "1" in all cases (not just bond) to simplify the test?

Reply:
We discussed it before and decided to use default num_paths value for the tests.

Reply:
can we remove the special case for bond interface?
in general it's better to avoid any special checks/cases and make tests as simple as possible